django-stubs/contrib/admin/options.pyi

```diff
@@ -93,35 +107,35 @@ class BaseModelAdmin:
     def has_module_permission(self, request: HttpRequest) -> bool: ...
 
 class ModelAdmin(BaseModelAdmin):
-    list_display: Sequence[Union[str, Callable]] = ...
+    list_display: Sequence[Union[str, Callable[[Model], Any]]] = ...
     list_display_links: Optional[Sequence[Union[str, Callable]]] = ...
     list_filter: Sequence[Union[str, Type[ListFilter], Tuple[str, Type[ListFilter]]]] = ...
     list_select_related: Union[bool, Sequence[str]] = ...
     list_per_page: int = ...
     list_max_show_all: int = ...
     list_editable: Sequence[str] = ...
     search_fields: Sequence[str] = ...
-    date_hierarchy: Optional[Any] = ...
+    date_hierarchy: Optional[str] = ...
     save_as: bool = ...
     save_as_continue: bool = ...
     save_on_top: bool = ...
-    paginator: Any = ...
+    paginator: Type = ...
     preserve_filters: bool = ...
     inlines: Sequence[Type[InlineModelAdmin]] = ...
-    add_form_template: Any = ...
-    change_form_template: Any = ...
-    change_list_template: Any = ...
-    delete_confirmation_template: Any = ...
-    delete_selected_confirmation_template: Any = ...
-    object_history_template: Any = ...
-    popup_response_template: Any = ...
-    actions: Any = ...
-    action_form: Any = ...
+    add_form_template: str = ...
+    change_form_template: str = ...
+    change_list_template: str = ...
+    delete_confirmation_template: str = ...
+    delete_selected_confirmation_template: str = ...
+    object_history_template: str = ...
+    popup_response_template: str = ...
+    actions: Sequence[Callable[[ModelAdmin, HttpRequest, QuerySet], None]] = ...
+    action_form: Any = ...  # TODO: Undocumented, remove?
```

Comment:
Let's leave all undocumented feature for now.

Reply:
Gotcha! 👍